From e92e54929d8736e472731c3365dcf57f8f505756 Mon Sep 17 00:00:00 2001
From: dengbo <bo.deng@spacemit.com>
Date: Wed, 8 Jan 2025 14:09:00 +0800
Subject: [PATCH 1430/1451] img-rogue: fix crash when use 'lsof /dev/video0'

fix:
kernel: Unable to handle kernel access to user memory without uaccess routines at virtual address 0000000000000018
kernel: Oops [#4]
kernel: Modules linked in: algif_hash algif_skcipher af_alg 8852bs binfmt_misc sch_fq_codel ip_tables autofs4
kernel: CPU: 5 PID: 5100 Comm: lsof Tainted: G      D            6.6.36 #2.0.4.2
kernel: Hardware name: spacemit k1-x deb1 board (DT)
kernel: epc : pvr_show_fdinfo+0x50/0x86
kernel: ra : seq_show+0x150/0x1a8
kernel: epc : ffffffff808259f6 ra : ffffffff802bb850 sp : ffffffc808e53c10
kernel: gp : ffffffff820f46e0 tp : ffffffd903ea1880 t0 : ffffffff80fd4f1c
kernel: t1 : ffffffff814aa8f0 t2 : ffffffff81401250 s0 : ffffffc808e53c60
kernel: s1 : ffffffd906ac9078 a0 : ffffffff80825822 a1 : ffffffc808e53c18
kernel: a2 : ffffffd902a96100 a3 : ffffffd906ac7080 a4 : 0000000000000000
kernel: a5 : ffffffd9044e8000 a6 : 0000000000000001 a7 : ffffffffffffffff
kernel: s2 : ffffffd907ff2700 s3 : ffffffd906ac9078 s4 : ffffffd906c679c0
kernel: s5 : ffffffd915e713d8 s6 : 0000000000088002 s7 : 0000000000000000
kernel: s8 : 0000000000400cc0 s9 : fffffffffffff000 s10: 000000007ffff000
kernel: s11: ffffffd906ac90b0 t3 : 0000003fa69ad9d8 t4 : 0000003fa6a4c790
kernel: t5 : 0000000000000016 t6 : ffffffd90f62302a
kernel: status: 0000000200000120 badaddr: 0000000000000018 cause: 000000000000000d
kernel: [<ffffffff808259f6>] pvr_show_fdinfo+0x50/0x86
kernel: [<ffffffff802bb850>] seq_show+0x150/0x1a8
kernel: [<ffffffff8025c8ac>] seq_read_iter+0xde/0x348
kernel: [<ffffffff8025cb92>] seq_read+0x7c/0xa8
kernel: [<ffffffff802326fc>] vfs_read+0x98/0x234
kernel: [<ffffffff8023307a>] ksys_read+0x54/0xcc
kernel: [<ffffffff80233106>] __riscv_sys_read+0x14/0x1c
kernel: [<ffffffff80fd4fd6>] do_trap_ecall_u+0xba/0x12c
kernel: [<ffffffff80fde792>] ret_from_exception+0x0/0x6e
kernel: Code: 67b0 0593 fb84 6fd8 7394 6a1c 3823 fc04 3423 fc94 (4f18) 52b4
kernel: ---[ end trace 0000000000000000 ]---
Change-Id: I26d8e12dd350d3bd20e10da517e102569bd0ab70
---
 drivers/gpu/drm/img-rogue/config_kernel.h  | 1 -
 drivers/gpu/drm/img-rogue/config_kernel.mk | 1 -
 2 files changed, 2 deletions(-)

diff --git a/drivers/gpu/drm/img-rogue/config_kernel.h b/drivers/gpu/drm/img-rogue/config_kernel.h
index 09747a8a950f..a29dbd592503 100644
--- a/drivers/gpu/drm/img-rogue/config_kernel.h
+++ b/drivers/gpu/drm/img-rogue/config_kernel.h
@@ -267,7 +267,6 @@
 #define PVRSRV_DEVICE_INIT_MODE PVRSRV_LINUX_DEV_INIT_ON_CONNECT
 #define SUPPORT_DI_BRG_IMPL
 #define SUPPORT_FW_CORE_CLK_RATE_CHANGE_NOTIFY
-#define SUPPORT_LINUX_FDINFO
 #define PVR_LINUX_PHYSMEM_MAX_POOL_PAGES 10240
 #define PVR_LINUX_PHYSMEM_ZERO_ALL_PAGES
 #define PVR_LINUX_PHYSMEM_MAX_EXCESS_POOL_PAGES 20480
diff --git a/drivers/gpu/drm/img-rogue/config_kernel.mk b/drivers/gpu/drm/img-rogue/config_kernel.mk
index c1c9e873d6a4..32eec2daf32f 100644
--- a/drivers/gpu/drm/img-rogue/config_kernel.mk
+++ b/drivers/gpu/drm/img-rogue/config_kernel.mk
@@ -53,5 +53,4 @@ override SUPPORT_WRAP_EXTMEM := 1
 override SUPPORT_NATIVE_FENCE_SYNC := 1
 override SUPPORT_DMA_FENCE := 1
 override SUPPORT_FW_CORE_CLK_RATE_CHANGE_NOTIFY := 1
-override SUPPORT_LINUX_FDINFO := 1
 override SUPPORT_BUFFER_SYNC := 1
-- 
2.47.0

